super-save.el

```diff
@@ -150,15 +156,17 @@ This function relies on the variable `super-save-predicates'."
 
 (defun super-save-command ()
   "Save the current buffer if needed."
-  (when (super-save-p)
-    (super-save-delete-trailing-whitespaces-maybe)
-    (if super-save-silent
-        (with-temp-message ""
-          (let ((inhibit-message t)
-                (inhibit-redisplay t)
-                (message-log-max nil))
-            (basic-save-buffer)))
-      (basic-save-buffer))))
+  (dolist (buf (if super-save-all-buffers (buffer-list) (list (current-buffer))))
+    (with-current-buffer buf
```

Comment:
Might be better to extract a `super-save-buffer` function, so you can separate more cleanly the looping from the process of saving a buffer.